dandiapi/api/tests/test_ingest_zarr_archive.py

```diff
@@ -0,0 +1,138 @@
+from typing import List
+
+import pytest
+
+from dandiapi.api.management.commands.ingest_zarr_archive import ingest_zarr_archive
+from dandiapi.api.models.zarr import ZarrArchive, ZarrUploadFile
+from dandiapi.api.zarr_checksums import (
+    ZarrChecksum,
+    ZarrChecksumFileUpdater,
+    ZarrChecksumListing,
+    ZarrChecksums,
+    ZarrChecksumUpdater,
+    ZarrJSONChecksumSerializer,
+)
+
+
+@pytest.mark.django_db
+def test_ingest_zarr_archive(zarr_upload_file_factory, zarr_archive_factory, faker):
+    zarr: ZarrArchive = zarr_archive_factory()
+
+    # Generate > 1000 files, since the page size from S3 is 1000 items
+    foo_bar_files: List[ZarrUploadFile] = [
+        zarr_upload_file_factory(zarr_archive=zarr, path='foo/bar/a'),
+        zarr_upload_file_factory(zarr_archive=zarr, path='foo/bar/b'),
+    ]
+    foo_baz_files: List[ZarrUploadFile] = [
+        zarr_upload_file_factory(zarr_archive=zarr, path=f'foo/baz/{faker.pystr()}')
+        for _ in range(1005)
```

Comment:
I'm not sure of the randomness of `pystr`, but it's generally a good idea to include the iterated value in the generated path to prevent collisions (`path=f'foo/baz/{i}_{faker.pystr()}'`). Especially with 1005 entries, you're likely to hit one eventually.